ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/rest/RestQueryTranslationTest.java

```diff
@@ -121,6 +127,17 @@ public void tearDown() {
     REST_APP.closePersistentQueries();
     REST_APP.dropSourcesExcept();
     TEST_HARNESS.getKafkaCluster().deleteAllTopics(TestKsqlRestApp.getCommandTopicName());
+
+    if (STARTING_THREADS.get() == null) {
```

Comment:
Very useful.  How did you discover the leak without this kind of test in place to begin with?

Reply:
I knew there was an issue because RQTT wouldn't run locally in the IDE for me.  Then investigated  with VisualVM :D